aten/src/ATen/native/AutogradComposite.cpp

```diff
@@ -22,6 +22,47 @@ std::tuple<at::Tensor, at::Tensor> _unpack_dual(const at::Tensor& tensor, int64_
   return std::tuple<at::Tensor, at::Tensor>(tensor._fw_primal(level), tensor._fw_grad(level));
 }
 
+Tensor _new_zeros_with_same_feature_meta(
+    const at::Tensor& self,
+    const at::Tensor& other,
+    int64_t self_num_batch_dims) {
```

Comment:
I would expect a check between self.dim() = other.dim() + self_num_batch_dims to make sure they are consistent?

Reply:
That is not true in the inplace over view case where the view changes the dim of the tensor

Reply:
Ho right self can be really anything. Ok